haskell/private/actions/link.bzl

```diff
@@ -381,11 +381,17 @@ def link_library_static(hs, cc, dep_info, objects_dir, my_pkg_id, with_profiling
             static_library,
             "@" + objects_dir_manifest.path,
         ])
+
+        if hs.toolchain.is_windows:
```

Comment:
@mboes I don't know how to get rid of this. Ideally, I'd like to set `hs.toolchain.tools.ar` as `cc.tools.ar` on anything but Windows ; but I can't find a way to get my hand the `cc` context within toolchain configuration.

Reply:
`cc_interop_info` in `cc.bzl` is the function creating the `cc` context. It gets called by rules that all have a `//haskell:toolchain` toolchain dependency. So you can get at `hs.tools.ar` from there. That said, I'm not convinced this is necessary. Are you sure `ar` fails if it's not the GHC one? `ar` is a very simple program, whose inputs are all explicit, and whose output format hasn't changed in years or even decades...

Reply:
Well, for starters we don't have a proper MSYS2 toolchain on the Windows CI. We could add one, but with current plans to tether Bazel's CC toolchain on GHC's on Windows, ( cf #304 ) it seems a bit silly…